plasma_framework/contracts/src/exits/payment/PaymentStandardExitable.sol

```diff
@@ -0,0 +1,153 @@
+    }
+
+    /**
+     * @notice Starts a standard withdrawal of a given output. Uses output-age priority.
+     * @dev requires the exiting UTXO's token to be added via 'addToken'
+     * @param _utxoPos Position of the exiting output.
+     * @param _outputTx RLP encoded transaction that created the exiting output.
+     * @param _outputTxInclusionProof A Merkle proof showing that the transaction was included.
+     */
+    function startStandardExit(
+        uint192 _utxoPos,
+        bytes calldata _outputTx,
+        bytes calldata _outputTxInclusionProof
+    )
+        external
+        payable
+        onlyWithValue(standardExitBond)
+    {
+        StartStandardExitData memory data = setupStartStandardExitData(_utxoPos, _outputTx, _outputTxInclusionProof);
+        verifyStartStandardExitData(data);
+        saveStandardExitData(data);
+        enqueueStandardExit(data);
+
+        emit ExitStarted(data.output.owner(), data.exitId);
+    }
+
+    /**
+    Private functions
+    */
+
+    function setupStartStandardExitData(
+        uint192 _utxoPos,
+        bytes memory _outputTx,
+        bytes memory _txInclusionProof
+    )
+        private
+        view
+        returns (StartStandardExitData memory)
+    {
+        UtxoPosLib.UtxoPos memory utxoPos = UtxoPosLib.UtxoPos(_utxoPos);
+        PaymentTransactionModel.Transaction memory outputTx = PaymentTransactionModel.decode(_outputTx);
+        PaymentOutputModel.Output memory output = outputTx.outputs[utxoPos.outputIndex()];
+        bool isTxDeposit = isDeposit.test(utxoPos.blockNum());
+        uint192 exitId = ExitId.getStandardExitId(isTxDeposit, _outputTx, utxoPos);
+        (bytes32 root, uint256 blockTimestamp) = framework.blocks(utxoPos.blockNum());
+
+        return StartStandardExitData({
+            utxoPos: utxoPos,
+            outputTx: outputTx,
+            output: output,
+            exitId: exitId,
+            isTxDeposit: isTxDeposit,
+            txInclusionProof: _txInclusionProof,
+            rlpTx: _outputTx,
+            txBlockRoot: root,
+            txBlockTimeStamp: blockTimestamp
+        });
+    }
+
+    function verifyStartStandardExitData(StartStandardExitData memory data)
+        private
+        view
+    {
+        bytes32 leafData = keccak256(data.rlpTx);
+        require(
+            Merkle.checkMembership(leafData, data.utxoPos.txIndex(), data.txBlockRoot, data.txInclusionProof),
+            "transaction inclusion proof failed"
+        );
+        require(data.output.amount > 0, "Should not exit with amount 0");
+        require(data.output.owner() == msg.sender, "Only output owner can start an exit");
+        require(exits[data.exitId].exitable == false, "Exit already started");
+    }
+
+    function saveStandardExitData(StartStandardExitData memory data) private {
+        PaymentExitDataModel.StandardExit memory exitData = PaymentExitDataModel.StandardExit({
```

Comment:
It  costs less gas if you don't use a temp variable e.g.
```
exits[data.exitId] =  PaymentExitDataModel.StandardExit({
...
```
This is because the `exits` array is in `storage` and using a temp `memory` variable forces a copy from `memory` to `storage`.
If you assign directly to the array then there's no need for the copy.

Reply:
@kevsul I'm going to merge this PR, as it's I need some code added here. In a couple minutes I'm going to make another PR that applies your comments.